src/urllib3/request.py

```diff
@@ -68,6 +69,7 @@ def request(
         body: Optional[HTTPBody] = None,
         fields: Optional[_TYPE_FIELDS] = None,
         headers: Optional[Mapping[str, str]] = None,
+        json: Optional[HTTPBody] = None,
```

Comment:
`HTTPBody` is not the correct type for this. It is possible to to provide some type safety with something like `Union[None, bool, float, str, Sequence[Any], Mapping[str, Any]]` but without support for recursive types in mypy it is not possible to provide an exact type (hence the Any's). So IMO, you can just go with `object`.

BTW, `None` is a valid input to `json.dumps` so you might want to use some sentinel object instead of `None` for the default.